app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+(function() {
+  angular.module('openshiftConsole').component('mobileServiceInstanceRow', {
+    controller: [
+      '$filter',
+      '$routeParams',
+      '$scope',
+      'APIService',
+      'AuthorizationService',
+      'BindingService',
+      'DataService',
+      'ListRowUtils',
+      MobileServiceInstanceRow
+    ],
+    controllerAs: 'row',
+    bindings: {
+        apiObject: '<',
+        serviceClass: '<',
+        state: '<'
+    },
+    templateUrl: 'views/_mobile-service-instance-row.html'
+  });
+
+  function MobileServiceInstanceRow($filter,
+                              $routeParams,
+                              $scope,
+                              APIService,
+                              AuthorizationService,
+                              BindingService,
+                              DataService,
+                              ListRowUtils) {
+    var row = this;
+    var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
+    var annotationSpace = "org.aerogear.binding." + this.apiObject.metadata.name;
+    var extendedAnnotationSpace = "org.aerogear.binding-ext." + this.apiObject.metadata.name;
+    var watches = [];
+    var bindingsWatch = false;
+    row.bindingsLimit = _.get(row.serviceClass, "spec.externalMetadata.bindingsLimit", 1);
+    _.extend(row, ListRowUtils.ui);
+
+    $scope.$on('$destroy', function(){
+        DataService.unwatchAll(watches);
+        DataService.unwatch(bindingsWatch);
+    });
+
+    watches.push(DataService.watchObject(APIService.getPreferredVersion('mobileclients'), $routeParams.mobileclient, {namespace: $routeParams.project}, function(mobileClient) {
+        row.mobileClient = mobileClient;
+
+        row.annotations = _.filter(row.mobileClient.metadata.annotations, function(annotation, name){
+          return name.split("/")[0] === annotationSpace;
+        })
+        .map(JSON.parse)
+        .sort(function(a, b) {
+          if (a.label < b.label) {
+            return -1;
+          }
+          if (a.label > b.label) {
+            return 1;
+          }
+          return 0;
+        });
+
+        row.serviceType = _.get(row, "serviceClass.spec.externalMetadata.serviceName"); // safe get
+
+        row.extendedAnnotations = _(row.mobileClient.metadata.annotations)
+          .pickBy(function(annotation, name){
+            return name.split("/")[0] === extendedAnnotationSpace;
+          })
+          .transform(function(result, strVal, name) {
+            var key = name.split("/")[1];
+            result[key] = JSON.parse(strVal);
+          })
+          .value();
+
+        row.servicePlanPromise = DataService.list(APIService.getPreferredVersion('clusterserviceplans'), {}, function(servicePlans) {
```

Comment:
You aren't passing to the view, so `row.servicePlanPromise` isn't needed.

Reply:
Do you need a fail case? If a user isn't allowed to see cluster scoped resources?

Reply:
`row.servicePlanPromise` will be removed.

Reply:
I am not sure about the call to `APIService.getPreferredVersion('clusterserviceplans')`. The bindService uses this APi to get the binding parameters so we did the same -> https://github.com/openshift/origin-web-console/blob/master/app/scripts/directives/bindService.js#L177

Is there an authorization for creating bindings I need to apply?